src/shared/store/index.js

```diff
@@ -1,10 +1,21 @@
-import { createStore, compose } from 'redux';
-import middlewareBuilder from './middlewareBuilder';
+import { createStore, compose} from 'redux';
+import createMiddlewares from './createMiddlewares';
 import rootReducer from './rootReducer';
+import devTools from 'remote-redux-devtools';
+import env from './../../shared/env';
 
 export default ({ initialState, history }) => {
-    const createStoreWithMiddleware = compose(...middlewareBuilder(history))(createStore);
-    const store = createStoreWithMiddleware(rootReducer, initialState);
+ if (instanceName === 'Client'){
+     debugger;
+ }
+    let enhancers = [createMiddlewares(history)];
+    if(env.nodeEnv !== 'production') {
+        enhancers = enhancers.concat([
+            typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : (f) => f,
+            devTools()
+        ]);
+    }
+    const store = createStore(rootReducer, initialState, compose(...enhancers));
```

Comment:
Wire up dev tools when not in production. Like how additional middlewares are considered enhancers. Good naming convention!